06.bowling_object/frame.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require_relative './shot'
+
+class Frame
+  TOTAL_FRAMES = 10
+
+  attr_reader :scores
+
+  def initialize(scores, frame_number)
+    @scores = scores
+    @frame_number = frame_number
+    @shots = generate_shots
+  end
+
+  def calculate_point(frame_scores)
+    @shots.sum(&:point) + calculate_additional_point(frame_scores)
+  end
+
+  def calculate_additional_point(frame_scores)
+    return 0 if @frame_number == TOTAL_FRAMES
+
+    shot_count = count_shot_count
+    score_index = frame_scores.take(@frame_number).flatten.size - 1
+    shots = (1..shot_count).map { |i| Shot.new(frame_scores.flatten[score_index + i]) }
+    shots.sum(&:point)
+  end
+
+  def count_shot_count
+    if strike?
+      2
+    elsif spare?
+      1
+    else
+      0
+    end
+  end
+
+  def strike?
+    @shots[0].point == 10
```

Comment:
```suggestion
    @shots[0].strike?
```

Shotクラス自身がstrikeかどうかを判断する、というアプローチもありそうです